### ISSUE TITLE
First questionnaire done, but the page doesn't go to the next experiment.

### ISSUE THREAD

**Issue by tnsgmddb** [expfactory/expfactory-experiments#228]
![2_no response](https://cloud.githubusercontent.com/assets/16784562/13055678/b1755bca-d3de-11e5-9616-47ef9669990c.png)

Object { rt: 4604, key_press: 13, exp_id: "dickman", trial_type: "text", trial_index: 0, time_elapsed: 4606, internal_node_id: "0.0-0.0" }

I finished the Dickman Questionnaire on a FireFox browser. I clicked the "Next Experiment" button, but nothing happens.

**Comment by tnsgmddb** [comment]
After the "poldrack-single-stim" , I clicked "Next Experiment", but nothing happens.

**Comment by tnsgmddb** [comment]
The "Next Experiment" button after the "poldrack-survey-multi-choice" questionnaire doesn't work to progress to the next experiment.

**Comment by tnsgmddb** [comment]
exp_id: "selection_optimization_compensation",

The "Next Experiment" button after the "selection_optimization_compensation" questionnaire doesn't work to progress to the next experiment.

**Comment by vsoch** [comment]
ok so to confirm - the "Next Experiment" button works on most experiments, but not in the case of the last JsPsych block being "poldrack-single-stim" or "poldrack-multi-source-button" - this can help to determine if it is a firefox or a jspsych or an experiment factory issue. I should note that I haven't done any testing with firefox (just chrome).

**Comment by vsoch** [comment]
has it worked for any experiments?

**Comment by vsoch** [comment]
I'm testing this now.

**Comment by tnsgmddb** [comment]
I am using Firefox and The Next Experiment never worked so far. 

Mary Ann is taking the tasks on Internet Explore.  She says the Next Experiment never worked for her either.

**Comment by vsoch** [comment]
Interesting, I just tried it on Firefox, and it worked for me. If you do:

      jsPsych.finishTrial()

and do that as many times as it takes to finish and then click the button, does it work? I will test Dickman next.

**Comment by vsoch** [comment]
and to be more clear, you type that in the console (right click --> inspect element --> Console tab)

**Comment by vsoch** [comment]
I'm having trouble debugging this, the experiments are working for me in Firefox. @IanEisenberg @zenkavi maybe you can give a try as well!

**Comment by IanEisenberg** [comment]
I'll look into it on firefox. I don't think it has to do with "poldrac-single-stim" or anything like that. I believe she was reporting the experiment it happened on, and accidentally reported the trial. The trial should have nothing to do with "next experiment" (in fact, it should always be proceeded by a "poldrack-text").

**Comment by tnsgmddb** [comment]
"I believe she was reporting the experiment it happened on, "  ====> yes

**Comment by vsoch** [comment]
The experiment could be an important variable in that if something in jsPsych breaks (eg, a bug is triggered and the standard functions aren't working as we assume) this would prevent the code from working at the end. However, I was not able to reproduce the lack of a functioning button testing with the specific experiment and browser. The other huge variable is Windows (I think we now all use Linux over here?) so we should find a way to test on Windows. @IanEisenberg do you still dual boot and have windows as an option?

**Comment by tnsgmddb** [comment]
I tried both "jsPsych.endExperiment" and "jsPsych.finishTrial() " 
![2_no response_console](https://cloud.githubusercontent.com/assets/16784562/13057392/e5f9c4ca-d3e9-11e5-9805-a59764f5553e.png)

I refreshed the page, then took  "exp_id: "grit_scale", and I am about to go to the Next Experiment . But the button doesn't do anything. 
![2_no response_console 2](https://cloud.githubusercontent.com/assets/16784562/13057436/288e2948-d3ea-11e5-91c9-7fb104e78b31.png)

**Comment by IanEisenberg** [comment]
I do have access to windows. I'll check it out.

**Comment by tnsgmddb** [comment]
Just FYI, This no progress issue continues after finishing the questionnaires, the Grit scale and " dospert_rt"

Thank you Vanessa and Ian for looking into this. I guess typing  jsPsych.endExperiment or jsPsych.finishTrial  into Console supposes to resolve this issue. But it seems not working for me,

**Comment by vsoch** [comment]
I think this is likely some bugs in the experiments themselves, and what I'm looking at is the error:

      timeline[current_experiment] is undefined

If you see that error, any typing of jsPsych.finishTrial() or jsPsych.endExperiment() would probably not work. Also, when you type either of these things, make sure to have "()" at the end, to indicate calling a function. Without it (as you see in your screen shots) you are displaying the function object in the window (vs. calling it).

**Comment by IanEisenberg** [comment]
It'd be odd for the experiments to have worked for Patrick and not now, but
definitely possible. Does this seem like a general questionnaire issue? Has
this happened after a task?

**Comment by vsoch** [comment]
I think it looks like a general questionnaire issue - another debug strategy to try is to end the experiment (way way of endExperiment() or finishTrial()) before the bug has a chance to trigger in the browser, and then see if the buttons work.

**Comment by tnsgmddb** [comment]
Vanessa when I followed your instruction about () I get these responses. Am I doing something wrong?

jsPsych.finishTrial()
TypeError: timeline[current_location] is undefined
 jsPsych.endExperiment()
undefined

screenshot attached below
![2_no response_console 3](https://cloud.githubusercontent.com/assets/16784562/13058049/5baf5a00-d3ee-11e5-9b5d-5d2d4c0b6f86.png)
.

**Comment by vsoch** [comment]
This indicates an error in the experiment so this function does not work.

**Comment by IanEisenberg** [comment]
There may be multiple issues going on. jsPsych.endExperiment() worked on IE for the task I was on (I don't claim it works all the time), but the "Next Experiment" button does not work ("Redo experiment" does).

I receive this error:
HTTP403: FORBIDDEN - The server understood the request, but is refusing to fulfill it.
(XHR): POST - http://expfactory.org/local/70/

**Comment by vsoch** [comment]
ah!

**Comment by vsoch** [comment]
It's an http/https issue. Have you tried using https?

**Comment by IanEisenberg** [comment]
Getting the same error:
HTTP403: FORBIDDEN - The server understood the request, but is refusing to fulfill it.
(XHR): POST - https://expfactory.org/local/71/

I don't even know why one was "https" and one was "http". I didn't do anything differently.

**Comment by vsoch** [comment]
Could you please test this link:

     http://www.expfactory.org/local/4/

both for http and https?

**Comment by IanEisenberg** [comment]
Seems to work fine for both.

{"message": "received!"}

In console:
HTML1300: Navigation occurred.
4

**Comment by vsoch** [comment]
ok this helps us to know it's not an issue with GET, but with POST, and likely the csrf token is not being properly passed. The weird thing that I'm not sure about is why it works for me (and sometimes for others?)

**Comment by IanEisenberg** [comment]
I have it work on chrome and firefox on linux and on chrome in Windows. It fails for Edge on Windows.

@tnsgmddb are you using firefox on windows?

**Comment by tnsgmddb** [comment]
Yes, Firefox on Windows

**Comment by zenkavi** [comment]
Sorry to be late to the game but is there more clarity on whether this is questionnaire specific issue that I should look in to in more detail?

**Comment by IanEisenberg** [comment]
This is not specific to questionnaires

On Tuesday, February 16, 2016, Zeynep Enkavi <notifications@github.com>
wrote:

> Sorry to be late to the game but is there more clarity on whether this is
> questionnaire specific issue that I should look in to in more detail?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/expfactory/expfactory-experiments/issues/228#issuecomment-184941026>
> .
>


-- 
Best,
Ian

**Comment by IanEisenberg** [comment]
@vsoch, have you figured out what the error is for this?

**Comment by IanEisenberg** [comment]
@tnsgmddb, is this still an issue? If you type "jsPsych.endExperiment()" (or get to the redo/next experiment screen naturally), can you progress?

I ask because I was able to on firefox/chrome on linux and Edge/Chrome on Windows. I know you are using firefox on Windows.

**Comment by tnsgmddb** [comment]
Yes I am using Firefox on Windows, and still the same issue occurs. I clicked "NEXT Experiment" but the page does not progress. I typed jsPsych.endExperiment()  .  
But I get this --> ReferenceError: jsPsych is not defined

I clicked Redo Experiment. Then I received a new task starting with the consent form. But the consent form (I mentioned this before #227 ) is in white background color.   
![1_consent-font color_white](https://cloud.githubusercontent.com/assets/16784562/13162377/992e1728-d672-11e5-9ff3-b296265134d0.png)

**Comment by IanEisenberg** [comment]
I have fixed the white background. That should only have happened before the hierarchical task, dietary decision, PRP, or dimensional shift.

Ok, so firefox on windows still buggy.

**Comment by tnsgmddb** [comment]
Yea, This white background consent is before the "hierarchical_rule" task

**Comment by vsoch** [comment]
When you get these errors about jsPsych not defined, it would be helpful to see the complete output in the console. Usually when buttons break and you see messages like there, there is some previous javascript error.

**Comment by tnsgmddb** [comment]
![javaerrors](https://cloud.githubusercontent.com/assets/16784562/13163424/de9f08b4-d67a-11e5-93ea-843f14a6c483.png)

@vsoch please see the screenshot. I had to reload the page, and then I got here, the first page of the "bickel_titrator" task. I pressed buttons but nothing happening. 

The messages in the console are as follows (shown in the screenshot attached): 

Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help http://xhr.spec.whatwg.org/
